proposed/2021/PackageNamespaces.md

```diff
@@ -0,0 +1,568 @@
+
+By providing an allowlist of package namespaces, we believe we can meet the security needs of the ecosystem from cybersquatting & typosquatting attacks like [Dependency Confusion](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610). Additionally, we can extend a security feature that makes NuGet unique of allowing [package creators to reserve a package ID prefix](https://docs.microsoft.com/nuget/nuget-org/id-prefix-reservation).
+
+This work will also allow future experiences in browsing, installing, and updating packages under an allowed list of namespaces to make developers secure by default when managing their dependencies in their software supply chain.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+When using a combination of public, private, and local sources defined in `NuGet.config` file(s), a user can add a new `<packageNamespaces>` element to opt-in to the feature. They can then create `<packageSource>` children elements to define the feed in which they'd like to add allowed namespaces to. Lastly by adding individual `<namespace>` elements in the `<packageSource>` node, the source will only allow the matching package namespaces from the respective package source.
+
+**Definition:**
+
+Add a new `<packageNamespaces>` element within the `NuGet.config` using the following syntax:
+
+```xml
+<packageSources>
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+         
+    <add key="contoso" value="https://contoso.com/packages/" />
+ 
+</packageSources>
+ 
+<packageNamespaces>
+</packageNamespaces>
+```
+
+Define the `<packageSource>` element within the `<packageNamespaces>` parent with the name of a valid package source:
+
+```xml
+<packageSources>
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+         
+    <add key="contoso" value="https://contoso.com/packages/" />
+ 
+</packageSources>
+ 
+<packageNamespaces>
+    <packageSource name="nuget.org">
```

Comment:
I think the attribute should probably be: 'key' to be consistent with disabledPackageSources and packageSources. It also indicates a tie to the packageSource element